staging/src/k8s.io/csi-translation-lib/translate_test.go

```diff
@@ -181,15 +190,15 @@ func TestTopologyTranslation(t *testing.T) {
 			t.Errorf("Expected node affinity %v, got %v", *test.expectedNodeAffinity, *nodeAffinity)
 		}
 
-		// Translate back to in-tree and make sure node affinity is still set
+		// Translate back to in-tree and make sure node affinity has been removed
```

Comment:
i am lacking some background, I am not sure when in practice TranslateCSIPVToInTree would be used, but in some theoretical rollback scenario where it is used, wouldn't it be better if the labels were put back? label -> node affinity -> label

Reply:
never mind, the labels would still be there, since the translation from in tree to csi preserved it and added node affinity. label != nodeAffinity

Reply:
From what I understand so far, TranslateCSIPVToInTree will be used in the csi-provisioner. After the CSI provisions the PV, the provisioner will translate it back to an intree format, and then create the PV.

If we do not remove the CSI topology, there will be rollback issues. For example, suppose we have a PV created when CSIMigration is turned on, this PV then will have a CSI topology node selector on it. And if the user somehow decide to rollback. That PV will become completely unusable because the CSINode will be all gone.

> wouldn't it be better if the labels were put back? label -> node affinity -> label

I am not sure if I understand this correctly. Currently our InTreetoCSI is to get the pv label and set them to node affinity. Are you suggesting when CSItoInTree we fetch the nodeaffinity and transfer it to PV labels and add them there? I think this sounds good to me. @msau42  what do you think?

Reply:
Thanks for explanation

> Are you suggesting when CSItoInTree we fetch the nodeaffinity and transfer it to PV labels and add them there? 

Yeah that's what I meant but actually it shouldn't be necessary since the labels will have been preserved during the initial translation. ( see latest comment, i commented within seconds of you)

Feel free to resolve this comment.

Reply:
lol yeah you are right it was already there. nvm